src/geo/transform.js

```diff
@@ -1567,7 +1524,7 @@ class Transform {
         this._constraining = true;
 
         // alternate constraining for non-Mercator projections
-        if (this.projection.name !== 'mercator') {
+        if (this.projection.isReprojectedInTileSpace) {
```

Comment:
I think the constraining for the globe probably needs some changes, because currently we can't pan the globe beyond the minimum and maximum longitude. I think it's not a blocker now and probably we could just fix this in a follow-up PR to reduce the changes in this one.

Reply:
Good catch, there was some overlapping logic that got mixed up during the rebase (projection.wrap and renderWorldCopies are almost the same but not quite). Should be fixed by d5bf4ce10c70ede4dc51222cdb92ab30dc1d9335.